client/components/data/query-jetpack-plugins/index.jsx

```diff
@@ -28,7 +29,10 @@ class QueryJetpackPlugins extends Component {
 	}
 
 	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( isEqual( nextProps.siteIds, this.props.siteIds ) ) {
+		if (
+			isEqual( nextProps.siteIds, this.props.siteIds ) ||
+			nextProps.forceRefresh === this.props.forceRefresh
```

Comment:
This change looks fine to me, but I can't see `forceRefresh` referenced elsewhere in the PR. What does this change do? Are we passing in `forceRefresh` to `QueryJetpackPlugins` anywhere? (I might have missed something obvious!)

Reply:
> What does this change do?

It's a hacky way of getting around `isEqual( nextProps.siteIds, this.props.siteIds )`. The siteIds barely ever change in a single Calypso session and therefore the query component will never perform another fetch to get the plugin list.

> Are we passing in forceRefresh to QueryJetpackPlugins anywhere? 

No, but we could! :P 

Once I got it working I stopped asking questions.

Reply:
@ramonjd just thinking, could this have an unintended side-effect that if the list of siteIds changes but `forceRefresh` remains unchanged, then it won't refresh, when it usually should? From taking a second look at this change, it seems like it'll result in fewer refreshes for fetching plugins, rather than more 🤔

Reply:
> From taking a second look at this change, it seems like it'll result in fewer refreshes for fetching plugins, rather than more

Hmmm. What do you think we could better here?

I was thinking that forceRefresh could work like:

```
const forceRefresh = needToRefresh ? true : false;
...
< QueryJetpackPlugins forceRefresh={ forceRefresh } />

```

Reply:
Should the logic be something like:

```
		if (
			isEqual( nextProps.siteIds, this.props.siteIds ) &&
			! nextProps.forceRefresh
		) {
```

That way, if `forceRefresh` is set, then it won't skip refreshing just because the siteIds match? The only problem there is you then need to know when to switch `forceRefresh` to `false` or it'll just keep refreshing when the props change (which mightn't be all that frequent, so maybe it's fine?)

I see in #46070 for notifications, they went with the approach of storing in state e.g. `didForceRefresh` and `setForceRefresh` so that it isn't constantly switched on to `true`. This might be a bit overkill for us, here.

Alternately, do we need this change to `<QueryJetpackPlugins` in this PR? I don't _think_ it's doing anything currently because we're not passing `forceRefresh` to it from the settings page, so perhaps it could be skipped? Or is this needed for a follow-up change? If we do need the change, then we might want to do a quick smoke test of the other places `<QueryJetpackPlugins>` is used, just to be sure.

Apologies if I'm being overly cautious! I just realised that I didn't _quite_ know as much about how this component is used as I thought I did when I started reviewing 😅

Reply:
You're right. It's best to be sure. 

>  I don't think it's doing anything currently because we're not passing forceRefresh to it from the settings page, so perhaps it could be skipped? Or is this needed for a follow-up change?

I wasn't planning on doing anything really. I did smoke test the other pages, e.g., `activity-log` and it still runs once over there.

> Alternately, do we need this change to <QueryJetpackPlugins in this PR?

We need it for the functionality in this PR to work at all. We could extract it into another PR, which would have to be merged first.

> I see in #46070 for notifications, they went with the approach of storing in state e.g. didForceRefresh and setForceRefresh so that it isn't constantly switched on to true. This might be a bit overkill for us, here.

Thanks for looking into that. It would be nice to have. Maybe in a follow up? Or do you think we should defend ourselves first?

Reply:
> We need it for the functionality in this PR to work at all.

Ah, gotcha. Because I couldn't see `forceRefresh` passed in anywhere, I wasn't sure if it was needed. If it gets it working, then I say go for it!

I think if it's enabled the PR to work the way we expect, and if it isn't causing any trouble for the other uses of `QueryJetpackPlugins`, then I'd just double-check that it's firing when you expect it to and merge it in. Any of the other improvements we can totally do as follow-ups if we run into any issues 👍

Thanks for your patience with this review!

Reply:
> Because I couldn't see forceRefresh passed in anywhere, I wasn't sure if it was needed. 

Andy, your reason strikes again... for the good of the world!

I don't need it. I was hallucinating. Thanks for such a thorough review and for discussing this.

Reverting, and merging without. 🙇